docs/content/ingestion/index.md

```diff
@@ -100,8 +100,9 @@ This is for realtime ingestion. Make sure to include "io.druid.extensions:druid-
 | type | String | This should say `avro_stream`. | no |
 | avroBytesDecoder | JSON Object | Specifies how to decode bytes to Avro record. | yes |
 | parseSpec | JSON Object | Specifies the format of the data. | yes |
+| schemaMappings | JSON Object | Specifies a mapping from field names found within the parseSpec to data within a complex avro structure. | no |
```

Comment:
Yes, what I proposed does need more work to do... and for the reasons you don't like it I think maybe there are some workrounds:

 1, I don't know how about other guys, but we need it in `JsonParser` to extract nested objects, maybe we didn't follow the best practice that using flattern task in ETL. Even for the parsers that need no schemaMappings, it still can fallback to old `List<String>dimensions`, so it's totally backwards compatible.

 2/3, If put the mapping into dimensions, it can be similar to `metricsSpec`. And
 for the prevalidation, it can steal spec grammer from pivot... Then it will be sth. like:
```json
"dimensionsSpec" : {
  "dimensionsMapping": [{
    "name" : "user",
    "fieldName" : "userName",
  },{
    "name" : "language",
    "expression" : "userInfo.language[0]"
  }],
  "dimensionExclusions" : [],
  "spatialDimensions" : []
}
```
If you keep use schemaMapping way, I think it should stay within Parsers instead of DimensionSpec, because it works for metrics extraction too.

For the record, I'm totally satisfied with the current method and just thinking about how we can add schemaMapping capability to JsonParser.